libtiledbsoma/src/version.cc

```diff
@@ -0,0 +1,52 @@
+/**
+ * @file   version.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This exposes the version of the TileDB Embedded library in use.
+ */
+
+#include "tiledbsoma/version.h"
+#include <tiledb/tiledb>
+#include "tiledbsoma/logger_public.h"
+
+#ifdef BUILD_COMMIT_HASH
+#define VERSION BUILD_COMMIT_HASH
+#else
+#define VERSION "dev"
+#endif
+
+namespace tiledbsoma::version {
+
+std::string as_string() {
+    int major, minor, patch;
+    tiledb_version(&major, &minor, &patch);
+    return fmt::format(
```

Comment:
This may get fmt::format linking issues.  Can you just use snprintf along these lines

```r
> Rcpp::cppFunction(r"(std::string v(int a, int b, int c) { char txt[32]; snprintf(txt, 32, "%d.%d.%d", a, b, c); return std::string(txt); })")
> v(2,3,4)
[1] "2.3.4"
> 
```

Reply:
it does work & this is just a move from `src/pyapi/libtiledbsoma.cc` where it also worked :)

Reply:
Well I think that is a little cavalier with respect to linking issues seesn with `libfmt`.  Note that Core does not call out to fmt / libfmt either.

Reply:
@eddelbuettel this is simply a move of existing code from one file to another. it works, as evidenced by CI output.